src/Features/Core/Portable/GenerateMember/GenerateVariable/AbstractGenerateVariableService.cs

```diff
@@ -216,10 +216,43 @@ private static void GenerateWriteableField(ArrayBuilder<CodeAction> result, Sema
                 if (containingMethod.Parameters.Length > 0)
                 {
                     var compilation = await document.Project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
-                    var lastParameterIsCancellationToken = Equals(containingMethod.Parameters.Last().Type, compilation.CancellationTokenType());
+                    var cancellationTokenType = compilation.CancellationTokenType();
 
-                    if (lastParameterIsCancellationToken)
-                        parameterIndex--;
+                    for (var i = containingMethod.Parameters.Length - 1; i >= 0; i--)
+                    {
+                        var parameter = containingMethod.Parameters[i];
+
+                        // Keep moving the insertion position for the generated parameter backwards
+                        // until we get to a parameter that does not need to be at the end of the
+                        // parameter list.
+                        if (parameter.HasExplicitDefaultValue)
+                        {
+                            parameterIndex--;
+                        }
+                        else if (parameter.IsParams)
+                        {
+                            parameterIndex--;
+                        }
+                        else if (parameter.RefKind is RefKind.Out)
+                        {
+                            parameterIndex--;
+                        }
+                        else if (Equals(parameter.Type, cancellationTokenType))
+                        {
+                            parameterIndex--;
+                        }
+                        else
+                        {
+                            break;
+                        }
```

Comment:
nit.  but personally this is clearer to me:

```suggestion
                        if (parameter.HasExplicitDefaultValue ||
                            parameter.IsParams ||
                            parameter.RefKind is RefKind.Out ||
                            Equals(parameter.Type, cancellationTokenType))
                        {
                            parameterIndex = i;
                            continue;
                        }
                        
                        break;
```

Effectively, the -- is a bit odd, it's like we're moving hte parameter index around.  instead, what we're really trying to say is: we know what index we're at.  as long as we see one of these cases, the inserted parameter should go before this parameter.